cmd/pull.go

```diff
@@ -0,0 +1,82 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/driusan/dgit/git"
+)
+
+func Pull(c *git.Client, args []string) error {
+	flags := flag.NewFlagSet("pull", flag.ExitOnError)
+	flags.SetOutput(flag.CommandLine.Output())
+	flags.Usage = func() {
+		flag.Usage()
+		fmt.Fprintf(flag.CommandLine.Output(), "\n\nOptions:\n")
+		flags.PrintDefaults()
+	}
+
+	fetchopts := git.FetchOptions{}
+
+	// These flags can be moved out of these lists and below as proper flags as they are implemented
+	for _, bf := range []string{"all", "a", "append", "unshallow", "update-shallow", "dry-run", "f", "force", "k", "keep", "multiple", "p", "prune", "P", "prune-tags", "n", "no-tags", "t", "tags", "no-recurse-submodules", "u", "update-head-ok", "q", "quiet", "v", "verbose", "progress", "4", "ipv4", "ipv6"} {
+		flags.Var(newNotimplBoolValue(), bf, "Not implemented")
+	}
+	for _, sf := range []string{"depth", "deepend", "shallow-since", "shallow-exclude", "refmap", "recurse-submodules", "j", "jobs", "submodule-prefix", "recurse-submodules-default", "upload-pack", "o", "server-option"} {
+		flags.Var(newNotimplStringValue(), sf, "Not implemented")
+	}
+
+	mergeopts := git.MergeOptions{}
+
+	flags.BoolVar(&mergeopts.FastForwardOnly, "ff-only", false, "Only allow fast-forward merges")
+	flags.BoolVar(&mergeopts.NoFastForward, "no-ff", false, "Create a merge commit even when it's a fast-forward merge.")
+
+	flags.Parse(args)
+
+	var repository string
+	var refspec []string
+	config, err := git.LoadLocalConfig(c)
+	if err != nil {
+		return err
+	}
+
+	if flags.NArg() < 1 {
+		// TODO simplistic, probably won't work in all cases
+		head := c.GetHeadBranch().BranchName()
+		branchremote, _ := config.GetConfig("branch." + head + ".remote")
+		repository = branchremote
+		refspec = []string{repository + "/" + head}
```

Comment:
1. my .git/config seems to have store the refspec in `branch.$BRANCHNAME.merge`
2. I don't think what you're storing in the refspec variable is an actual refspec. (refspecs are things that live under the refs/ subdirectory of $GIT_DIR.)

Reply:
There's some very mysterious stuff happening in there. For example, many of the refs in that config value for my repos don't point to remote refs in the refs/remotes/... directory. Instead, they seem to point back to local branches, which doesn't seem to make sense. Also, in my repos that just have a single "origin" repo and are on the master branch there are no .git/refs/remotes/origin/master refs. Instead, there's just the .gif/refs/remotes/origin/HEAD files, which seem to point back at themselves.

Reply:
I think it's the name *on the remote* to send to `git fetch-pack`, not the local ref name.  We probably can't do it properly until fetch-pack is implemented, but you can add a comment and rename the variable for now